pop_tools/fill.py

```diff
@@ -141,45 +179,127 @@ def _iterative_fill_POP_core(nlat, nlon, var, fillmask, missing_value, tol, ltri
                 denom = 0.0
 
                 # East
-                if var[j, ip1] != missing_value:
+                if np.isfinite(var[j, ip1]):
                     numer += var[j, ip1]
                     denom += 1.0
 
                 # North
                 if j < nlat - 1:
-                    if var[jp1, i] != missing_value:
+                    if np.isfinite(var[jp1, i]):
                         numer += var[jp1, i]
                         denom += 1.0
 
                 else:
                     # assume only tripole has non-land top row
                     if ltripole:
-                        if var[j, nlon - i] != missing_value:
-                            numer += var[j, nlon - i]
+                        if np.isfinite(var[j, nlon - 1 - i]):
+                            numer += var[j, nlon - 1 - i]
```

Comment:
I have verified that the test that is currently failing (`test_lateral_fill_np_array_ltripole` in `tests/test_fill.py`) passes if I revert these two lines back to the old values. Clearly we want the updates since this fixes an indexing bug that results in invalid indices when `i=0`.

@andersy005 I looked at the test code itself, and it looks like the solution is to update `lateral_fill_np_array_tripole_filled_ref.npz` in

```py3
    # load reference data
    ref_data_file = DATASETS.fetch('lateral_fill_np_array_tripole_filled_ref.npz')
    with np.load(ref_data_file) as data:
        z_fill_ref = data['arr_0']
```

How do we do that?

Reply:
@mnlevy1981, the original file resides here `/home/abanihi/ftp/aletheia-data/test-data` on hobart. By update, you mean updating the original file, If you point to some file on hobart, I can put it on the ftp server.

Reply:
> @mnlevy1981, the original file resides here `/home/abanihi/ftp/aletheia-data/test-data` on hobart. If by update you mean updating the original file, If you point me to some file on hobart, I can put it on the ftp server.

Cool, is there an easy way to generate the file? (Is it already being generated by the test and I just need to find it?)

Reply:
> Is it already being generated by the test and I just need to find it

The original file was provided by @matt-long. I don't recall seeing how it was generated. Ccing @matt-long

Reply:
This is a regression test. I think the test itself documents how the file was created. The process for recreating this file would be to apply the same algorithm as the test, verify the results by inspection, and write a new file.

Reply:
@andersy005 

> If you point me to some file on hobart, I can put it on the ftp server.

I generated `/home/mlevy/for_anderson/lateral_fill_np_array_tripole_filled_ref.npz` by adding

```py3
np.savez_compressed('lateral_fill_np_array_tripole_filled_ref.npz', z_fill)
```

However, I haven't been able to test this file because if I copy it into `.pop_tools/data/` then `pooch` over-writes it with the old version of the file. I tried testing by hand by reading in the old version of the file and then the new version of the file, and got an unexpected dimensions mismatch:

```py3
>>> np.testing.assert_allclose(new['arr_0'], old['arr_0'], atol=1e-14, equal_nan=True, verbose=True)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/mlevy/miniconda3/envs/pop-tools/lib/python3.8/site-packages/numpy/testing/_private/utils.py", line 1527, in assert_allclose
    assert_array_compare(compare, actual, desired, err_msg=str(err_msg),
  File "/Users/mlevy/miniconda3/envs/pop-tools/lib/python3.8/site-packages/numpy/testing/_private/utils.py", line 759, in assert_array_compare
    raise AssertionError(msg)
AssertionError:
Not equal to tolerance rtol=1e-07, atol=1e-14

(shapes (41, 81), (42, 82) mismatch)
 x: array([[     nan,      nan,      nan, ...,      nan,      nan,      nan],
       [0.20738 ,      nan,      nan, ...,      nan,      nan,      nan],
       [0.234307, 0.320153,      nan, ...,      nan,      nan,      nan],...
 y: array([[     nan,      nan,      nan, ...,      nan,      nan,      nan],
       [     nan,      nan,      nan, ...,      nan,      nan,      nan],
       [0.127922, 0.20738 ,      nan, ...,      nan,      nan,      nan],...
```

comparing that output to the failing test output, it looks like `new['arr_0']` is missing the first row and first column? There should be an additional row of `nan`s, and the `0.20738` at the beginning of the second row should be the second element of the third row... but I don't see this shape mismatch when actually running the test, which is a little confusing.

Reply:
@andersy005 I'm not sure what was happening in the above comment with the weird shape, but I checked out `master` and verified the test generated a `(42, 82)` array and then when I went back to the branch from this PR it was still generating that size array rather than the erroneous `(41, 81)` array I had seen an hour earlier. So I'm more confident that `/home/mlevy/for_anderson/lateral_fill_np_array_tripole_filled_ref.npz` is the right file for the FTP server.

Reply:
Great... I just updated the file on the FTP server with your new copy. For this to work properly, you will need to update the sha256 entry in https://github.com/NCAR/pop-tools/blob/d771f8d7280696c890bb6a21255c5ad298c8b373/pop_tools/data_registry.txt#L7


to this new value:

`a6e5ac70da6762d8423dda3675fe03230ff8a55dc47bc15d6f47c1c53ca9e2d9`

Reply:
Is there some lag before the FTP server is updated? My tests are still pulling down the old file, and looking at the [ftp site](https://bit.ly/3h9oJyr) (apologies for the bit.ly url, github issues apparently doesn't render ftp links correctly) in a browser, it looks like the file was last modified January 1st

Reply:
>  github issues apparently doesn't render ftp links correctly) in a browser, it looks like the file was last modified January 1st

Well, something is definitely not right:

```bash
In [19]: !ls -lah
total 50M
drwxr-sr-x 2 abanihi cgdaletheia 3.9K Aug 18 15:46 .
drwxrwsr-x 8 mclong  cgdaletheia 3.9K Apr  8 21:02 ..
-rw-r--r-- 1 abanihi cgdaletheia  49M Jan  2  2020 cesm_pop_monthly.T62_g17.nc
-rw-r--r-- 1 abanihi cgdaletheia  20K Aug 18 16:54 lateral_fill_np_array_filled_ref.npz
-rw-r--r-- 1 abanihi cgdaletheia  20K Aug 18 16:57 lateral_fill_np_array_tripole_filled_ref.npz
```


What is the output of 

```python
In [5]: import pop_tools

In [6]: pop_tools.DATASETS.registry
```

Reply:
Per #58 I think we should add `/home/mlevy/for_anderson/lateral_fill_np_array_tripole_filled_ref.20200818.npz` to the server and leave `lateral_fill_np_array_tripole_filled_ref.npz` untouched; then the update to `data_registry.txt` would be to add the line

```
lateral_fill_np_array_tripole_filled_ref.20200818.npz a6e5ac70da6762d8423dda3675fe03230ff8a55dc47bc15d6f47c1c53ca9e2d9
```

Instead of modifying the existing entry
```

Reply:
Done! 

@mnlevy1981, I restored the original files, and added a copy of the new `lateral_fill_np_array_tripole_filled_ref.20200818.np` file to the FTP server.